sdk/trace/span.go

```diff
@@ -286,3 +304,43 @@ func startSpanInternal(name string, parent core.SpanContext, remoteParent bool,
 
 	return span
 }
+
+type samplingDecision struct {
+	noParent     bool
+	remoteParent bool
+}
+
+type samplingData struct {
+	parent core.SpanContext
+	name   string
+	cfg    *Config
+	span   *span
+}
+
+type makeDecisionData func() samplingData
+
+func (d samplingDecision) makeDecision(f makeDecisionData) {
+	if d.noParent || d.remoteParent {
+		// If this span is the child of a local span and no
+		// Sampler is set in the options, keep the parent's
+		// TraceOptions.
+		//
+		// Otherwise, consult the Sampler in the options if it
+		// is non-nil, otherwise the default sampler.
+		data := f()
+		sampler := data.cfg.DefaultSampler
+		//if o.Sampler != nil {
+		//	sampler = o.Sampler
+		//}
+		spanContext := &data.span.spanContext
+		sampled := sampler(SamplingParameters{
+			ParentContext:   data.parent,
+			TraceID:         spanContext.TraceID,
+			SpanID:          spanContext.SpanID,
+			Name:            data.name,
+			HasRemoteParent: d.remoteParent}).Sample
+		if sampled {
+			spanContext.TraceOptions = core.TraceOptionSampled
```

Comment:
spanContext is in span and span.data. only one of them is updated here.
Also need to add test for
1. Intiailly span is not sampled but after set name it is.
2. vice-versa.